lib/datadog/appsec/contrib/rack/gateway/request.rb

```diff
@@ -41,7 +41,7 @@ def method
 
             def headers
               result = request.env.each_with_object({}) do |(k, v), h|
-                h[k.gsub(/^HTTP_/, '').downcase!.tr('_', '-')] = v if k =~ /^HTTP_/
+                h[k.gsub(/^HTTP_/, '').tap(&:downcase!).tap { |s| s.tr!('_', '-') }] = v if k =~ /^HTTP_/
```

Comment:
The test indicates that you are keeping all of the values, therefore in my opinion:

1. There should also be a test for mixed-case string input, e.g. HTTP_Foo
2. I think the logic as currently proposed will result in the mixed case being preserved which I think is not the right behavior

Reply:
As you pointed in your second comment, HTTP header names are uppercased by Rack (as HTTP header names are case-insensitive according to RFC 2616) and Rack will also drop empty headers so 'HTTP_' should not be possible either. But there's the case of someone adding a middleware to Rack that changes the headers for exemple. I believe that in these case, we still want to send everything to the WAF, as an attack could still be present in the value of these headers.